src/charm.py

```diff
@@ -356,6 +356,31 @@ def _block_on_openstack_config(self, state: CharmState) -> bool:
             return True
         return False
 
+    def _common_install_code(self, state: CharmState) -> bool:
```

Comment:
May I suggest moving this to perhaps `lxd.py`? I think that will help us start separating openstack implementations to lxd implementations for the refactor :D

Reply:
I think `lxd.py` is more about abstracting low-level lxd commands. Have you and @yhaliaw thought about where to put the openstack related charm code? I feel it is better to discuss this first before I move it somewhere and it is out of scope for this PR.